### ISSUE TITLE
job_options with 'constraint' gives InvalidJobDescriptionException

### ISSUE THREAD

**Issue by jhidding** [xenon-middleware/xenon-grpc#38]
Running a job from PyXenon with
```python
scheduler.submit_interactive_job(
    ....,
    options={'constraint': 'TitanX'})
```
gives the following GRPC error:
```
(StatusCode.FAILED_PRECONDITION, nl.esciencecenter.xenon.schedulers.InvalidJobDescriptionException: slurm adaptor: Given Job option "constraint" not supported)
```
Without the given job option, the job runs successfully.

**Comment by jhidding** [comment]
similarly, using the xenon-cli interface:
```
xenon scheduler slurm --location ssh://fs0.das5.cs.vu.nl/home/jhidding --username jhidding --certfile ~/.ssh/id_rsa --prop xenon.adaptors.schedulers.ssh.strictHostKeyChecking=false exec --option constraint=TitanX cat
slurm adaptor: Given Job option "constraint" not supported
```

**Comment by sverhoeven** [comment]
Setting a constraint like that is not supported until https://github.com/NLeSC/Xenon/issues/582 is implemented.

See https://github.com/NLeSC/Xenon/issues/556 for allowed options

**Comment by sverhoeven** [comment]
Use scheduler_arguments to specify constraints like ` --constraint="TitanX"`

Fixed in 5fd47190d6df7f2a30261d93cf7fb9db05634021
